packages/next/server/render.tsx

```diff
@@ -1364,20 +1387,30 @@ export async function renderToHTML(
       return { docProps, documentCtx }
     }
 
-    const renderContent = () => {
+    const renderContent = ({
+      EnhancedApp,
+      EnhancedComponent,
+    }: {
+      EnhancedApp?: AppType
+      EnhancedComponent?: NextComponentType
+    } = {}) => {
       return ctx.err && ErrorDebug ? (
         <Body>
           <ErrorDebug error={ctx.err} />
         </Body>
       ) : (
         <Body>
           <AppContainerWithIsomorphicFiberStructure>
-            {isServerComponent && !!AppMod.__next_rsc__ ? (
-              // _app.server.js is used.
-              <Component {...props.pageProps} />
-            ) : (
-              <App {...props} Component={Component} router={router} />
-            )}
+            {isServerComponent
+              ? React.createElement(EnhancedComponent || Component, {
+                  App: EnhancedApp || App,
```

Comment:
is `App` a required prop for Component?

Reply:
Yes it's passed in so the flight renderer can use it: https://github.com/vercel/next.js/pull/36352/files#diff-63632e6e580e69692941c89bab0a0e0436ade3f9543ef610925f795928d9d5fdR421